machine.go

```diff
@@ -653,8 +654,9 @@ type cloudInstance struct {
 
 	InstanceId_ string `yaml:"instance-id"`
 
-	Status_        *status `yaml:"status"`
-	StatusHistory_ `yaml:"status-history"`
+	Status_             *status `yaml:"status"`
+	StatusHistory_      `yaml:"status-history"`
+	ModificationStatus_ *status `yaml:"modification-status,omitempty"`
```

Comment:
I feel like these need nice doc strings clarifying what statuses end up where, and why someone would want to use one.

Reply:
Yes, will do